### ISSUE TITLE
alternative for-loop should be warned against

### ISSUE THREAD

**Issue by kvantour** [koalaman/shellcheck#2037]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
for i in 1 2 3; { echo "$i"; }
for i; { echo "$i"; }

```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
A warning or potentially an error should be raised for KSH and BASH and for sure an error should be raised for SH.
The usage of the do ... done construct should be suggested. Eg.

SCxxxx : alternative for-loop construct not recommended
The usage of a brace-group instead of do ... done is an undocumented feature and is recommended not be used.

The alternative for-loop is a feature of shell-implementations which can be traced back to the unix Bourne Shell V7.
This feature has never been documented as it should never have existed. But it is implemented to be compatible with very old-code. The main developer of bash (Chet Ramey) mentions that this feature should never be used and will never be documented. A bit more detail can be found on the following pages:

* https://stackoverflow.com/questions/63247449/alternative-for-loop-construct
* https://lists.gnu.org/archive/html/bug-bash/2020-08/msg00058.html

**Comment by kvantour** [comment]
alternative for-loop constructs are still accepted in the current online version of shell-check.
